test/e2e/upgrade_management_cluster_test.go

```diff
@@ -2,340 +2,137 @@ package e2e
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"os/exec"
-	"path/filepath"
-	"runtime"
 	"strings"
 
-	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/utils/pointer"
-	clusterv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4"
-	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
-	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
 	framework "sigs.k8s.io/cluster-api/test/framework"
-	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var _ = Describe("When testing cluster upgrade [upgrade]", func() {
-	upgradeManagementCluster()
-})
-
-func upgradeManagementCluster() {
-	BeforeEach(func() {
-		osType := strings.ToLower(os.Getenv("OS"))
-		Expect(osType).ToNot(Equal(""))
-		validateGlobals(specName)
-
-		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
-		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
+const workDir = "/opt/metal3-dev-env/"
+
+var _ = Describe("When testing cluster upgrade v1alpha5 > current [upgrade]", func() {
+	capi_e2e.ClusterctlUpgradeSpec(ctx, func() capi_e2e.ClusterctlUpgradeSpecInput {
+		return capi_e2e.ClusterctlUpgradeSpecInput{
+			E2EConfig:                 e2eConfig,
+			ClusterctlConfigPath:      clusterctlConfigPath,
+			BootstrapClusterProxy:     bootstrapClusterProxy,
+			ArtifactFolder:            artifactFolder,
+			SkipCleanup:               true, // parameterize after merging https://github.com/kubernetes-sigs/cluster-api/pull/7373
+			InitWithProvidersContract: "v1alpha4",
+			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.4.7/clusterctl-{OS}-{ARCH}",
```

Comment:
I think here we are testing upgrade from `v1alpha4` (capi) so setting this in the environment somewhere then propagate it to here with some logic doesn't seem needed so far there is not case where `InitWithProvidersContract` will be `v1beta1` and since the support for this api versoin already dropped I do not think we will get patches for `v0.4.x` so `v0.4.7` is the latest.

It is hardcoded but it makes debugging the input easier although we can move them to e2e read them from there

Reply:
@mboukhalfa you already have env var for them: `CAPI_FROM_RELEASE`

Reply:
Ok maybe v1alpha4 is fine but don't we already have the release here: https://github.com/metal3-io/cluster-api-provider-metal3/blob/6b3b651f81b54a48c9f39d5d402baf58c26af225/scripts/ci-e2e.sh#L62

Reply:
done